homeassistant/components/skybell/binary_sensor.py

```diff
@@ -16,11 +19,21 @@
 
 SCAN_INTERVAL = timedelta(seconds=10)
 
-# Sensor types: Name, device_class, event
-SENSOR_TYPES = {
-    "button": ["Button", DEVICE_CLASS_OCCUPANCY, "device:sensor:button"],
-    "motion": ["Motion", DEVICE_CLASS_MOTION, "device:sensor:motion"],
-}
+
+class SensorTypeClass(NamedTuple):
+    """Metadata for sensor types."""
+
+    name: str
+    # Sensor types: Name, device_class, event
+    button: list[str]
+    motion: list[str]
+
+
+SENSOR_TYPES = SensorTypeClass(
+    "sensor type",
+    button=["Button", DEVICE_CLASS_OCCUPANCY, "device:sensor:button"],
+    motion=["Motion", DEVICE_CLASS_MOTION, "device:sensor:motion"],
+)
```

Comment:
I think you mixed up where the NamedTuple should be used. The goal was to replace the list of strings so we don't  need index lookup anymore, not to replace the dict. As a good example, take a look at #53242

I you have any questions, please don't hesitate to ask, here or on discord.

Reply:
Updated to replace the array.